pkg/plugins/shell/command.go

```diff
@@ -0,0 +1,68 @@
+package shell
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+type command struct {
+	Cmd string
+	Dir string
+	Env []string
+}
+
+type commandResult struct {
+	ExitCode int
+	Stdout   string
+	Stderr   string
+}
+
+type commandExecutor interface {
+	ExecuteCommand(cmd command) (commandResult, error)
+}
+
+type nativeCommandExecutor struct {
+}
+
+func (nce *nativeCommandExecutor) ExecuteCommand(inputCmd command) (commandResult, error) {
+	if inputCmd.Cmd == "" {
```

Comment:
Good point, even if this is only a YAML scalar value (not a key, not a complex structure). I'm not sure though: I fear that it would risk having side effects if there is a `grep` with these characters within. I don't think I'll do this in this PR: either waiting for an issue to be raised with a use case, or maybe while adding an "array" form for specifying the spec.command (instead of the actual string).